controllers/topology/internal/mergepatch/mergepatch_test.go

```diff
@@ -100,6 +100,89 @@ func TestNewHelper(t *testing.T) {
 			wantHasSpecChanges: false,
 			wantPatch:          []byte("{\"metadata\":{\"labels\":{\"foo\":\"bar\"}}}"),
 		},
+		{
+			name: "Field (metadata.label) preserve instance specific values when path is not authoritative",
+			original: &unstructured.Unstructured{ // current
+				Object: map[string]interface{}{
+					"metadata": map[string]interface{}{
+						"labels": map[string]interface{}{
+							"a": "a",
+							"b": "b-changed",
+						},
+					},
+				},
+			},
+			modified: &unstructured.Unstructured{ // desired
+				Object: map[string]interface{}{
+					"metadata": map[string]interface{}{
+						"labels": map[string]interface{}{
+							// a missing
+							"b": "b",
+							"c": "c",
+						},
+					},
+				},
+			},
+			wantHasChanges:     true,
+			wantHasSpecChanges: false,
+			wantPatch:          []byte("{\"metadata\":{\"labels\":{\"b\":\"b\",\"c\":\"c\"}}}"),
+		},
+		{
+			name: "Field (metadata.label) align to modified when path is authoritative",
+			original: &unstructured.Unstructured{ // current
+				Object: map[string]interface{}{
+					"metadata": map[string]interface{}{
+						"labels": map[string]interface{}{
+							"a": "a",
+							"b": "b-changed",
+						},
+					},
+				},
+			},
+			modified: &unstructured.Unstructured{ // desired
+				Object: map[string]interface{}{
+					"metadata": map[string]interface{}{
+						"labels": map[string]interface{}{
+							// a missing
+							"b": "b",
+							"c": "c",
+						},
+					},
+				},
+			},
+			options:            []HelperOption{AuthoritativePaths{contract.Path{"metadata", "labels"}}},
+			wantHasChanges:     true,
+			wantHasSpecChanges: false,
+			wantPatch:          []byte("{\"metadata\":{\"labels\":{\"a\":null,\"b\":\"b\",\"c\":\"c\"}}}"),
+		},
+		{
+			name: "IgnorePaths supersede AuthoritativePaths",
+			original: &unstructured.Unstructured{ // current
+				Object: map[string]interface{}{
+					"metadata": map[string]interface{}{
+						"labels": map[string]interface{}{
+							"a": "a",
+							"b": "b-changed",
+						},
+					},
+				},
+			},
+			modified: &unstructured.Unstructured{ // desired
+				Object: map[string]interface{}{
+					"metadata": map[string]interface{}{
+						"labels": map[string]interface{}{
+							// a missing
+							"b": "b",
+							"c": "c",
+						},
+					},
+				},
+			},
+			options:            []HelperOption{AuthoritativePaths{contract.Path{"metadata", "labels"}}, IgnorePaths{contract.Path{"metadata", "labels"}}},
```

Comment:
Note: in our prod code we're using `...metadata` as authoritative code. Depending on the outcome of the discussion what we will use there, would be nice to have test cases which cover a few scenarios with the path/paths we're using

Reply:
The specific use case are already tested in `reconcile_state_test.go`